onnxruntime/core/session/inference_session.cc

```diff
@@ -1043,14 +1084,22 @@ common::Status InferenceSession::Initialize() {
 
 #if !defined(ORT_MINIMAL_BUILD)
     if (!session_options_.optimized_model_filepath.empty()) {
-      // Serialize optimized ONNX model.
-      ORT_RETURN_IF_ERROR_SESSIONID_(Model::Save(*model_, session_options_.optimized_model_filepath));
       if (session_options_.graph_optimization_level >= TransformerLevel::Level3) {
-        LOGS(*session_logger_, WARNING) << "Serializing Optimized ONNX model with Graph Optimization"
-                                           " level greater than ORT_ENABLE_EXTENDED. The generated"
-                                           " model may contain hardware and execution provider specific"
-                                           " optimizations, and should only be used in the same environment"
-                                           " the model was optimized for.";
+        LOGS(*session_logger_, WARNING)
+            << "Serializing optimized model with Graph Optimization level greater than ORT_ENABLE_EXTENDED. "
+               "The generated model may contain hardware and execution provider specific optimizations, "
+               "and should only be used in the same environment the model was optimized for.";
```

Comment:
Do we want to stop here if the optimized output is ORT? Most likely save ORT model is running offline on different hardware than mobile

Reply:
I don't think so. If we stopped we prevent someone saving an optimized model even if they explicitly want this level of optimization. Up to the user to choose if they want to change the optimization level. 

---
In reply to: [481381851](https://github.com/microsoft/onnxruntime/pull/4995#discussion_r481381851) [](ancestors = 481381851)